### PR TITLE
Bumps snakeyaml to 2.0

### PR DESCRIPTION
**Pull request by owaiskazi19**
### Description
Snakeyaml released the version [`2.0`](https://bitbucket.org/snakeyaml/snakeyaml/wiki/Changes) which also addresses the CVE-2022-1471. 

### Issues Resolved
https://github.com/opensearch-project/OpenSearch/issues/5576

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -34,6 +34,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Bump `joni` from 2.1.44 to 2.1.45
 - Bump `commons-io:commons-io` from 2.7 to 2.11.0
 - Bump `org.jruby.joni:joni` from 2.1.45 to 2.1.48
+- Bump `snakeyaml` from 1.33 to 2.0 ([#6511](https://github.com/opensearch-project/OpenSearch/pull/6511))
```
Comment (saratvemulapalli) [inline CHANGELOG.md:37]: Do you plan to backport this change to 2.x?

**Review by saratvemulapalli — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (andrross) [reply CHANGELOG.md:37]: Yes, please do backport (and move this to the 2.x section in this PR prior to the backport).

Comment (owaiskazi19) [reply CHANGELOG.md]: Pushed the change!

**Review by andrross — APPROVED** [review]

**Comment by andrross** [comment]
@owaiskazi19 Can you rebase this?

**Review by andrross — APPROVED** [review]

**Comment by kotwanikunal** [comment]
Backport failed due to changelog divergence. Fix: #6518

**Comment by cwperks** [comment]
@owaiskazi19 @saratvemulapalli Should this be backported to 1.3?

**Pull request merged**